django/db/models/sql/query.py

```diff
@@ -2300,16 +2357,29 @@ def add_deferred_loading(self, field_names):
         # splitting and handling when computing the SQL column names (as part of
         # get_columns()).
         existing, defer = self.deferred_loading
+
+        flatten_fields = set()
+        for field_name in field_names:
+            expanded = (field_name,)
+            if isinstance(field_name, str):
+                try:
+                    expanded = self.expand_components(field_name)
+                except FieldError:
+                    pass
+            if expanded is None:
+                expanded = (field_name,)
```

Comment:
This can only happen if the `try:` succeeds, so we can do:

```suggestion
                else:
                    if expanded is None:
                        expanded = (field_name,)
```

I think this way reduces the mental load slightly - my first reaction reading the code is "how can `expanded` be `None`? We set it to a tuple above".